src/common/telemetry/inc/Telemetry.h

```diff
@@ -0,0 +1,176 @@
+#else
+#define VERBOSE_FLAG_IF_DEBUG ""
+#endif
+
+// Initialize telemetry file - call once at program start (thread-safe)
+#define OSConfigTelemetryInit() { \
+    if (!g_telemetryFileInitialized) { \
+        pthread_mutex_lock(&g_telemetryMutex); \
+        if (!g_telemetryFileInitialized && !g_telemetryFile) { \
+            g_moduleDirectory = getModuleDirectory(); \
+            g_fileName = generateRandomFilename(); \
+            g_telemetryFile = fopen(g_fileName, "a"); \
+            if (g_telemetryFile) { \
+                g_telemetryFileInitialized = 1; \
+            } \
+        } \
+        pthread_mutex_unlock(&g_telemetryMutex); \
+    } \
+}
+
+// Cleanup telemetry file - call on program exit
+#define OSConfigProcessTelemetryFile() { \
+    if (g_telemetryFile && g_fileName && g_moduleDirectory) { \
+        char* command = FormatAllocateString("%s/%s %s %s %d", g_moduleDirectory, TELEMETRY_BINARY_NAME, VERBOSE_FLAG_IF_DEBUG, g_fileName, TELEMETRY_TIMEOUT_SECONDS); \
+        if (NULL != command) { \
+            ExecuteCommand(NULL, command, false, false, 0, TELEMETRY_TIMEOUT_SECONDS, NULL, NULL, NULL); \
+        } \
+        FREE_MEMORY(command); \
+    } \
+    g_telemetryFile = NULL; \
+    g_telemetryFileInitialized = 0; \
+    FREE_MEMORY(g_fileName); \
+    FREE_MEMORY(g_moduleDirectory); \
+}
+
+// Helper macro to append raw JSON string to telemetry file
+#define OSConfigTelemetryAppendJSON(jsonString) { \
+    if (!g_telemetryFileInitialized) { \
+        OSConfigTelemetryInit(); \
+    } \
+    if (g_telemetryFile) { \
+        fprintf(g_telemetryFile, "%s\n", jsonString); \
+        fflush(g_telemetryFile); \
+    } \
+}
+
+#define OSConfigTelemetryStatusTrace(callingFunctionName, status) \
+    OSConfigTelemetryStatusTraceImpl((callingFunctionName), (status), __LINE__)
+#define OSConfigTelemetryStatusTraceImpl(callingFunctionName, status, line) { \
+    char status_str[MAX_INT_STRING_LENGTH] = {0}; \
+    snprintf(status_str, sizeof(status_str), "%d", (status)); \
+    char line_str[MAX_INT_STRING_LENGTH] = {0}; \
+    snprintf(line_str, sizeof(line_str), "%d", (line)); \
+    const char* distroName = GetOsName(NULL); \
+    const char* correlationId = getenv(TELEMETRY_CORRELATIONID_ENVIRONMENT_VAR); \
+    const char* ruleCodename = getenv(TELEMETRY_RULECODENAME_ENVIRONMENT_VAR); \
+    const char* timestamp = GetFormattedTime(); \
+    char* telemetry_json = FormatAllocateString("{\"EventName\":\"StatusTrace\",\"Timestamp\":\"%s\",\"Filename\":\"%s\",\"LineNumber\":\"%s\",\"FunctionName\":\"%s\",\"RuleCodename\":\"%s\",\"CallingFunctionName\":\"%s\",\"ResultCode\":\"%s\",\"DistroName\":\"%s\",\"CorrelationId\":\"%s\",\"Version\":\"%s\"}", timestamp ? timestamp : "", __FILE__, line_str, __func__, ruleCodename ? ruleCodename : "", (callingFunctionName) ? (callingFunctionName) : "-", status_str, distroName ? distroName : "unknown", correlationId ? correlationId : "", OSCONFIG_VERSION); \
```

Comment:
>"{\"EventName\":\"StatusTrace\",\"Timestamp\":\"%s\",\"Filename\":\"%s\",\"LineNumber\":\"%s\",\"FunctionName\":\"%s\",\"RuleCodename\":\" [](http://example.com/codeflow?start=48&length=138)

You know that we can break such long string constants in many sub-strings, still forming same one string, yes?

Also, why not move these super long constants to their own constant macros and invoke those here?